src/EditorFeatures/Core/Shared/Utilities/IThreadingContextExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+
+internal static class IThreadingContextExtensions
+{
+    /// <summary>
+    /// Returns true if any keyboard or mouse button input is pending on the message queue.
+    /// </summary>
+    public static bool IsInputPending()
```

Comment:
static helper in ForegroundThreadAffinitizedObject that is still used i na few places.  not sure if we still need it, but keeping it around.